numpy/linalg/linalg.py

```diff
@@ -532,7 +532,7 @@ def inv(a):
     return wrap(ainv.astype(result_t, copy=False))
 
 
-def matrix_power(M, n):
+def matrix_power(a, n):
```

Comment:
In principle changing the argument name is a compatibility break - perhaps one we can get away with , but maybe worth a remark in the release notes. Maybe mention that it now throws `LinalgError` too?